stock_release_channel_shipment_advice/models/stock_release_channel.py

```diff
@@ -23,6 +23,9 @@ class StockReleaseChannel(models.Model):
         string="Shipment Advices",
         readonly=True,
     )
+    dock_id = fields.Many2one(
```

Comment:
@sbejaoui Do we need a constrains on `dock_id` and `warehouse_id` to avoid case where you set a warehouse then a dock and then unset the warehouse... In this case you'll end with a `dock_id` without a `warehouse`?

Reply:
@sbejaoui I confirm that if I remove the warehouse, the dock_id is still set....

Reply:
the planner checks that the dock belong to the warehouse, IMO, we don't need a constrains at the release channel level. But a onchange to unset the dock at the warehouse change make sense

Reply:
@sbejaoui  an onchange will not prevent to create inconsistencies when creating records directly from the code or xml data files. To avoid this kind of trouble, you could add an inverse on the warehouse_id to unset the dock_id if wharehouse_id is None

Reply:
please check my last commit